src/core/choroplethFeature.js

```diff
@@ -0,0 +1,245 @@
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.choropleth = function (arg1, arg2) {
+    if (arg1 === undefined) {
+      return m_choropleth;
+    }
+    if (typeof arg1 === 'string' && arg2 === undefined) {
+      return m_choropleth[arg1];
+    }
+    if (arg2 === undefined) {
+      var choropleth = $.extend(
+        {},
+        m_choropleth,
+        arg1
+      );
+      m_choropleth = choropleth;
+    } else {
+      m_choropleth[arg1] = arg2; //if you pass in accessor for prop
+    }
+    m_this.modified();
+    return m_this;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A uniform getter that always returns a function even for constant values.
+   * If undefined input, return all the choropleth values as an object.
+   *
+   * @param {string|undefined} key
+   * @return {function}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.choropleth.get = function (key) {
+    if (key === undefined) {
+      var all = {}, k;
+      for (k in m_choropleth) {
+        if (m_choropleth.hasOwnProperty(k)) {
+          all[k] = m_this.choropleth.get(k);
+        }
+      }
+      return all;
+    }
+    return geo.util.ensureFunction(m_choropleth[key]);
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A method that adds a polygon feature to the current layer.
+   *
+   * @param {array} coordinateArray
+   * @param {geo.color} fillColor
+   * @return {geo.feature}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this._addPolygonFeature = function(coordinateArray, fillColor){
+    return m_this.layer()
+      .createFeature('polygon', {
+      })
+      .data([{
+        type: 'Polygon',
+        coordinates: coordinateArray
+      }])
+      .polygon(function (d) {
+        return {
+          'outer': d.coordinates[0],
+          'inner': d.coordinates[1] // undefined but ok
+        };                              
+      })
+      .position(function(d){
+        return {
+          x: d[0],
+          y: d[1]
+        };
+      })
+      .style({
+        'fillColor': fillColor
+      });
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A method that adds polygons from a given feature to the current layer.
+   *
+   * @param {} geoJsonFeature
+   * @param geo.color
+   * @return [{geo.feature}] 
+   */
+  ////////////////////////////////////////////////////////////////////////////          
+  this._featureToPolygons = function (feature, fillValue) {
+
+    var arrayOfPolygonFeatures = [];
+    
+    if (feature.geometry.type === 'Polygon'){
+      arrayOfPolygonFeatures = [
+        m_this
+          ._addPolygonFeature(feature.geometry.coordinates, fillValue)
+      ];
+      
+    }
+    if (feature.geometry.type === 'MultiPolygon') {
```

Comment:
But "data" is already plural ;).

My issue is not with the implementation containing multiple objects, it's
naming the entire collection with the word "feature".

On Thu, Sep 3, 2015 at 11:54 AM Aashish Chaudhary <notifications@github.com>
wrote:

> In src/core/choroplethFeature.js
> <https://github.com/OpenGeoscience/geojs/pull/434#discussion_r38662226>:
>
> > +   * @param geo.color
> > +   * @return [{geo.feature}]
> > +   */
> > +  ////////////////////////////////////////////////////////////////////////////
> > +  this._featureToPolygons = function (feature, fillValue) {
> > +
> > +    var arrayOfPolygonFeatures = [];
> > +
> > +    if (feature.geometry.type === 'Polygon'){
> > +      arrayOfPolygonFeatures = [
> > +        m_this
> > +          ._addPolygonFeature(feature.geometry.coordinates, fillValue)
> > +      ];
> > +
> > +    }
> > +    if (feature.geometry.type === 'MultiPolygon') {
>
> I've always felt the word "feature" isn't being used quite correctly here.
> To me, it's more like a "layer" that contains several "features" (i.e.,
> "point feature" to me sounds like a single point on the map, not a bunch of
> points, bound to data, etc.).
>
> @ronichoudhury <https://github.com/ronichoudhury> it is typical in GIS
> and Graphics packages not to use plural. For example vtkPolyData would
> contain more than one point or lines or polygons. The idea is that a layer
> can contain multiple features (for efficiency or need point of view). Also,
> having multiple data points inside a feature is efficient from rendering
> point of view.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenGeoscience/geojs/pull/434/files#r38662226>.
>